src/ui/handler/drag_pan.js

```diff
@@ -89,13 +89,16 @@ class DragPanHandler {
     }
 
     _onDown(e: MouseEvent | TouchEvent) {
-        if (this._ignoreEvent(e)) return;
+        if (this._map.boxZoom.isActive()) return;
+        if (this._map.dragRotate.isActive()) return;
         if (this.isActive()) return;
 
         if (e.touches) {
+            if ((e.touches: any).length > 1) return;
             window.document.addEventListener('touchmove', this._onMove);
             window.document.addEventListener('touchend', this._onUp);
         } else {
+            if (e.ctrlKey || e.button !== 0) return;
```

Comment:
is `e.button` guaranteed to be defined in this case?  (Asking because in `_ignoreEvent`, we checked `e.button && e.button !== 0`.)